vllm/model_executor/layers/sampler.py

```diff
@@ -313,6 +322,28 @@ def _apply_top_p_top_k(
     return logits
 
 
+def _apply_min_p(
+    logits: torch.Tensor,
+    min_ps: List[float],
+) -> torch.Tensor:
+    min_p = torch.tensor(min_ps, dtype=logits.dtype, device=logits.device)
+    probs = torch.softmax(logits, dim=-1)
+    top_probs, _ = probs.max(dim=-1, keepdim=True)
+    scaled_min_p = min_p.unsqueeze(dim=1) * top_probs
+    tokens_to_remove = probs < scaled_min_p
+
+    sorted_indices = torch.argsort(logits, descending=True, dim=-1)
+    sorted_indices_to_remove = torch.gather(tokens_to_remove,
+                                            dim=-1,
+                                            index=sorted_indices)
```

Comment:
Seems like the sort here is unnecessary? I believe we only need to mask out the positions where `probs < scaled_min_p`?